roles/ohpc_add_rabbitmq_agents/tasks/main.yaml

```diff
@@ -1,41 +1,44 @@
 ---
 - name: Clone the git repo to the machine
   git: 
-    repo: https://github.com/uabrc/rabbitmq_agents.git
-    dest: /opt/rabbitmq_agents
+    repo: "{{ rabbitmq_agents_repo }}"
+    dest: "{{ rabbitmq_agents_loc }}"
     version: develop
 
 - name: Make a copy of rabbit config example file
-  copy:
-    src: /opt/rabbitmq_agents/rabbit_config.py.example
-    dest: /opt/rabbitmq_agents/rabbit_config.py
-    remote_src: yes
-
-- name: Add rabbitmq password in the rabbit_config file
-  replace:
-    path: /opt/rabbitmq_agents/rabbit_config.py
-    regexp: 'CHANGE_IT_TO_YOUR_OWN_PASSWORD'
-    replace: '{{ rabbitmq_user_password }}'
-    backup: yes
+  template:
+    src: config.j2
+    dest: "{{ rabbitmq_agents_loc }}/rabbit_config.py"
 
 - name: Check if environment is properly setup, and install python packages if not
   pip:
-    requirements: /opt/rabbitmq_agents/requirements.txt
-    executable: pip3
-
-- name: Run the ohpc_account_create agent in the background
-  shell: cd /opt/rabbitmq_agents; nohup python ohpc_account_create.py </dev/null >/dev/null 2>&1 & 
-
-#- name: Run the slurm_account_create agent in the background
-#  shell: cd /opt/rabbitmq_agents; nohup python slurm_agent.py </dev/null >/dev/null 2>&1 &
+    requirements: requirements.txt
+    virtualenv: venv
+    virtualenv_command: /usr/bin/python3 -m venv
+    chdir: "{{ rabbitmq_agents_loc }}"
 
 - name: Copy service files to the machine.
-  copy:
-    src: "{{ item }}"
-    dest: "/etc/systemd/system/{{ item }}"
-  with_items:
-    - ohpc_account.service
-    - slurm_account.service
+  template:
+    src: "{{ item }}.j2"
+    dest: "/etc/systemd/system/{{ item }}.service"
+  loop:
+    - ohpc_account
+    - slurm_account
```

Comment:
Should we add the suffix _agent to these so their function is a bit clearer.  I'm thinking we should refactor to "user_reg_agents" instead of rabbitmq_agents but don't want that change part of this PR.

Reply:
I also remember that you mentioned about role name. Should I change it to `ohpc_user_reg_agents`?

Reply:
fixed in b8ae8f2